arelle/ModelXbrl.py

```diff
@@ -20,7 +21,10 @@
 from arelle.ValidateXbrlDimensions import isFactDimensionallyValid
 
 
-ModelRelationshipSet = None # dynamic import
+
+if TYPE_CHECKING:
+    from arelle.ModelRelationshipSet import ModelRelationshipSet
```

Comment:
These imports are on separate paths so it works, but it's a bit awkward that `ModelRelationshipSet` refers to two separate things (the class here and the module below). I'd actually use your `ModelRelationshipSetClass` name from before here

```suggestion
    from arelle.ModelRelationshipSet import ModelRelationshipSet as ModelRelationshipSetClass
```

nit: also too many blank lines above